### PR TITLE
JS: Support for TypeScript 4.1

### PR DESCRIPTION
**Pull request by asgerf**
Currently just adds support for template literal types

### CODE REVIEW

**Comment by asgerf** [comment]
@erik-krogh can you review?

**Review by erik-krogh — COMMENTED** [review]
A comment about the `yarn.lock` file. 

The rest looks good to me.

File: javascript/extractor/lib/typescript/yarn.lock [hunk 1]
```diff
@@ -6,6 +6,7 @@
   version "12.7.11"
   resolved node-12.7.11.tgz#be879b52031cfb5d295b047f5462d8ef1a716446
 
-typescript@4.0.2:
-  version "4.0.2"
-  resolved "https://registry.yarnpkg.com/typescript/-/typescript-4.0.2.tgz#7ea7c88777c723c681e33bf7988be5d008d05ac2"
+typescript@4.1.2:
+  version "4.1.2"
+  resolved "https://registry.yarnpkg.com/typescript/-/typescript-4.1.2.tgz#6369ef22516fe5e10304aae5a5c4862db55380e9"
+  integrity sha512-thGloWsGH3SOxv1SoY7QojKi0tc+8FnOmiarEGMbd/lar7QOEd3hvlx3Fp5y6FlDUGl9L+pd4n2e+oToGMmhRQ==
```
[the remark this answers is not in the document]
Comment (asgerf) [reply javascript/extractor/lib/typescript/yarn.lock:12]: This has no effect on whether it uses the local copy or not (really). And we should not manually edit the lockfile. Been there done that; never again.

**Review by erik-krogh — APPROVED** [review]
LGTM.  

Optional: 
I think the workaround I added earlier can be removed now: https://github.com/github/codeql/blob/203bbdd84f2dfac38c4efd70df7d46632e13d289/javascript/extractor/src/com/semmle/ts/extractor/TypeScriptASTConverter.java#L1430-L1433

**Review by erik-krogh — APPROVED** [review]

**Review by erik-krogh — APPROVED** [review]

**Pull request merged**